src/kv/kv.h

```diff
@@ -119,11 +133,27 @@ namespace kv
       local_hook(local_hook_),
       global_hook(global_hook_)
     {
-      roll->push_back({0, State(), Write()});
+      roll->insert_back(CreateNewLocalCommit(0, State(), Write()));
     }
 
     Map(const Map& that) = delete;
 
+    template <typename... Args>
+    LocalCommit* CreateNewLocalCommit(Args&&... args)
```

Comment:
can these not be initialized as safe pointers? do we need raw pointers here?

Reply:
this would mean having dllist (part of snmalloc) support shared_ptr for next&prev pointers and add the cost of modifying atomics during the list operations.